src/spritecache.cpp

```diff
@@ -803,7 +803,7 @@ static void CompactSpriteCache()
 			GetSpriteCache(i)->ptr = s->data; // Adjust sprite array entry
 			/* Swap this and the next block */
 			temp = *s;
-			memmove(s, next, next->size);
+			std::move(next, &next[next->size], s);
```

Comment:
Not sure whether the overlapping side is known here, but `next` is `MemBlock *`, so `&next[next->size]` is a lot more than `next->size`.

Reply:
Given `NextBlock` uses the `size` variable to jump to the next block, it's only going forward, i.e. `next > s`.